### PR TITLE
GEOPY-1767: Support Value map with bytes

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-1767 - Support Value map with bytes**

### CODE REVIEW

File: geoh5py/data/reference_value_map.py [hunk 1]
```diff
@@ -65,12 +66,19 @@ def validate_value_map(cls, value_map: np.ndarray | dict) -> np.ndarray:
             if not np.all(np.asarray(list(value_map)) >= 0):
                 raise KeyError("Key must be an positive integer")
 
-            value_map = np.array(list(value_map.items()), dtype=cls.MAP_DTYPE)
+            value_list = list(value_map.items())
+
+            if isinstance(value_list[0][1], bytes):
```
[the remark this answers is not in the document]
Comment (MatthieuCMira) [reply geoh5py/data/reference_value_map.py:71]: Again, why not always converting bytes to string by default?

**Review by MatthieuCMira — COMMENTED** [review]
Just some questions. 
Maybe I'm not understanding every logic

File: geoh5py/io/h5_reader.py [hunk 2]
```diff
@@ -401,10 +401,12 @@ def fetch_type_attributes(cls, type_handle: h5py.Group) -> dict:
...
-            type_attributes["value_map"] = type_handle["Value map"][:].astype(
-                [("Key", "<u4"), ("Value", "<U13")]
-            )
-
+            try:
+                type_attributes["value_map"] = type_handle["Value map"][:].astype(
+                    [("Key", "<u4"), ("Value", "<U13")]
+                )
+            except UnicodeDecodeError:
+                type_handle["Value map"][:].astype([("Key", "<u4"), ("Value", "O")])
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoh5py/io/h5_reader.py:409]: Yikes, yes.

File: geoh5py/data/reference_value_map.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (domfournier) [reply geoh5py/data/reference_value_map.py:71]: I think so yea, will be simpler. Just convert to byte, then back to long string. Looks like GA doesn't care anyway

**Review by MatthieuCMira — DISMISSED** [review]
Good to me

File: geoh5py/data/reference_value_map.py [hunk 3]
```diff
@@ -42,7 +43,12 @@ def __len__(self) -> int:
         return len(self._map)
 
     def __call__(self) -> dict:
-        return dict(self._map)
+        try:
```
Comment (MatthieuCMira) [inline geoh5py/data/reference_value_map.py:46]: raising an error to catch it is not the best practice.

Maybe we could just convert to "O".
Or check if string, then convert.

**Review by MatthieuCMira — COMMENTED** [review]
just a small comment

**Pull request merged**